src/error-table.rkt

```diff
@@ -1,20 +1,460 @@
+           
+           ;; Hallucination:
+           ;; if x is large enough that x^y overflows, the condition number also
+           ;; is very large, but the answer correctly overflows
+           [(and (> y 1)
+                 (infinite? x^y)
+                 (infinite? subexpr-val)) #f]
+
+           ;; if x is small enough and y is large enough that x^y underflows,
+           ;; the condition number also gets very large, but the answer
+           ;; correctly underflows
+           [(and (> y 1)
+                 (zero? x^y)
+                 (zero? subexpr-val)) #f]
+
+           [(and (or (zero? x)
+                     (infinite? x))
+                 (not (= subexpr-val x^y)))
+            (mark-erroneous! subexpr)]
+            
+           [(or (bf> cond-x cond-thres)
+                (bf> cond-y cond-thres))
+            (mark-erroneous! subexpr)]
+           
+           [else #f])]
+        
+        [(list (or 'acos.f64 'acos.f32) x-ex)
+         #:when (list? x-ex)
+         (define x (exacts-ref x-ex))
+         (define x.bf (bf x))
+         (define cond-x (bfabs
+                         (bf/ x.bf
+                              (bf* (bfsqrt
+                                    (bf- 1.bf
+                                         (bf* x.bf x.bf)))
+                                   (bf subexpr-val)))))
+         
+         (cond
+           ; Condition number hallucinations:
+           ; acos(1) == 0
+           [(and (= x 1.0) (= subexpr-val 0.0)) #f]
+
+           ; acos(-1) == pi
+           [(and (= x -1.0) (= subexpr-val pi)) #f]
+           
+           ; High Condition Number:
+           ; CN(acos, x) = |x / (√(1 - x^2)acos(x))|
+           [(bf> cond-x cond-thres)
+            (mark-erroneous! subexpr)]
+           
+           [else #f])]
+
+        [(list (or 'asin.f64 'asin.f32) x-ex)
+         #:when (list? x-ex)
+         (define x (exacts-ref x-ex))
+         (define x.bf (bf x))
+         (define cond-x (bfabs
+                         (bf/ x.bf
+                              (bf* (bfsqrt
+                                    (bf- 1.bf
+                                         (bf* x.bf x.bf)))
+                                   (bf subexpr-val)))))
+         
+         (cond
+           ; Condition Number hallucinations:
+           ; asin(1) == pi/2
+           [(and (= x 1.0) (= subexpr-val (/ pi 2.0))) #f]
+
+           ; asin(-1) == -pi/2
+           [(and (= x -1.0)
+                 (= subexpr-val (/ (- pi) 2.0))) #f]
 
-  (for/hash ([group (in-list (group-by identity pt-worst-subexprs))])
-    (values (first group) (length group))))
+           ; asin(0) == 0
+           [(and (= x 0.0) (= subexpr-val 0.0)) #f]
+           
+           ; High Condition Number:
+           ; CN(acos, x) = |x / (√(1 - x^2)asin(x))|
+           [(bf> cond-x cond-thres) (mark-erroneous! subexpr)]
+           [else #f])]
+        [_ #f]))
+    (unless has-errored?
+      (mark-erroneous! #f)))
```

Comment:
I think if we're counting both overpredictions and underpredictions, we don't need the `#f` row?